adapters/repos/db/index.go

```diff
@@ -1092,26 +1104,33 @@ func (i *Index) exists(ctx context.Context, id strfmt.UUID,
 		return i.replicator.Exists(ctx, cl, shardName, id)
 	}
 
-	if shard := i.localShard(shardName); shard != nil {
-		exists, err = shard.Exists(ctx, id)
+	if !i.isLocalShard(shardName) {
```

Comment:
I think for read requests, we need to call getOrInitLocalShard() only when receiving a remote request from the coordinator, like the calls of this kind:` IncomingB***()`. For other read requests, we might just need to check if the local shard exists (e.g., `objectByID()` and similar). What do you think?

Reply:
`isLocalShard` implementation changed not to call leader. If shard is not inited locally, request will be passed to same node through remote call and shard will be inited/created by `IncomingXYZ` method